stock_customer_deposit_sale_margin/models/sale_order_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.depends(
+        "product_id",
+        "company_id",
+        "currency_id",
+        "product_uom",
+        "deposit_available_qty",
+    )
+    def _compute_purchase_price(self):
+        res = super()._compute_purchase_price()
+        for line in self.filtered_domain([("deposit_available_qty", ">", 0.0)]):
+            # Set purchase price to zero because purchase price is in sale order
+            # that made customer deposit, otherwise the cost would be posted twice.
+            line.purchase_price = 0.0
```

Comment:
Maybe check if deposit available is < than demanded quantity?

```suggestion
        "deposit_available_qty",
        "product_uom_qty",
    )
    def _compute_purchase_price(self):
        res = super()._compute_purchase_price()
        for line in self.filtered_domain([("deposit_available_qty", ">", 0.0)]):
            # Set purchase price to zero because purchase price is in sale order
            # that made customer deposit, otherwise the cost would be posted twice.
            if line.deposit_available_qty < line.product_uom_qty:
                continue
            line.purchase_price = 0.0
```

Reply:
I think this is not necessary, user can't confirm sale if customer deposit is lower than qty demanded. See https://github.com/OCA/stock-logistics-workflow/blob/16.0/stock_customer_deposit/models/sale_order.py#L61

Reply:
Makes sense, indeed, because if the uom qty is higher, then it can't be borrowed from the deposit.

Beware, your suggestions has a bug IIUC: it should be `<=`, not just `<`. Also you can optimize it a bit more by looping only once:

```suggestion
        "deposit_available_qty",
        "product_uom_qty",
    )
    def _compute_purchase_price(self):
        res = super()._compute_purchase_price()
        for line in self:
            if 0 < line.deposit_available_qty <= line.product_uom_qty:
                continue
            # Set purchase price to zero because purchase price is in sale order
            # that made customer deposit, otherwise the cost would be posted twice.
            line.purchase_price = 0.0
```